--- a/selenium_driverless/types/context.py
+++ b/selenium_driverless/types/context.py
@@ -434,7 +434,7 @@
     @property
     async def window_handles(self) -> List[TargetInfo]:
         """Returns the handles of all windows within the current session."""
-        warnings.warn("window_handles aren't ordered")
+        # warnings.warn("window_handles aren't ordered")
         tabs = []
         targets = await self.targets
         for info in list(targets.values()):

