--- a/selenium_driverless/types/target.py
+++ b/selenium_driverless/types/target.py
@@ -833,7 +833,7 @@
         context_id = None
         # noinspection PyProtectedMember
         if self._context._is_incognito:
-            context_id = self.browser_context_id
+            context_id = await self.browser_context_id
         return await add_cookie(target=self, cookie_dict=cookie_dict, context_id=context_id)
 
     @property
